### PR TITLE
Drop dead beta storage and validation code

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Drops the storage and validation code that was only exercised by the beta versions no longer served since 1.22.

#### Does this PR introduce a user-facing change?
```release-note
kube-apiserver: The `rbac.authorization.k8s.io/v1alpha1` API version is removed; use the `rbac.authorization.k8s.io/v1` API, available since v1.8. The `scheduling.k8s.io/v1alpha1` API version is removed; use the `scheduling.k8s.io/v1` API, available since v1.14.
```

/assign @deads2k

### CODE REVIEW

File: pkg/apis/certificates/validation/validation.go [hunk 1]
```diff
@@ -420,24 +418,19 @@ func findConditions(csr *certificates.CertificateSigningRequest, conditionType c
 // compatible with the specified version and existing CSR.
 // oldCSR may be nil if this is a create request.
 // validation options related to subresource-specific capabilities are set to false.
-func getValidationOptions(version schema.GroupVersion, newCSR, oldCSR *certificates.CertificateSigningRequest) certificateValidationOptions {
+func getValidationOptions(newCSR, oldCSR *certificates.CertificateSigningRequest) certificateValidationOptions {
 	return certificateValidationOptions{
-		allowResettingCertificate:    allowResettingCertificate(version),
```
Comment (deads2k) [inline pkg/apis/certificates/validation/validation.go:425]: I didn't know we ever allowed this.  weird.

Comment (liggitt) [reply pkg/apis/certificates/validation/validation.go:425]: accidentally, in v1beta1, like so much else

**Comment by deads2k** [comment]
Now that's a cathartic PR.

/lgtm

**Comment by liggitt** [comment]
/assign @robscott 
PTAL at the `Drop legacy validation logic for networking API` commit

/assign @msau42 
PTAL at the `Drop legacy status logic for volumeattachments API` commit

**Comment by msau42** [comment]
/lgtm
for volumeattachment validation

**Comment by tengqm** [comment]
Why https://github.com/kubernetes/kubernetes/tree/master/pkg/apis/rbac/v1alpha1 is still there and https://github.com/kubernetes/kubernetes/tree/master/pkg/apis/scheduling/v1alpha1 is still there if both APIs are dropped?

**Comment by liggitt** [comment]
All code related to *serving* those versions was removed, but the types and conversions still exist to support reading/converting data **stored in etcd** in those versions

**Comment by tengqm** [comment]
So ... those schemes are still installed, just for conversions convenience?

**Comment by tengqm** [comment]
@liggitt Thanks for the clarification.

**Pull request merged**